cpp/oneapi/dal/algo/louvain/common.cpp

```diff
@@ -0,0 +1,65 @@
+/*******************************************************************************
+* Copyright 2021 Intel Corporation
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*******************************************************************************/
+
+#include "oneapi/dal/algo/louvain/common.hpp"
+#include "oneapi/dal/detail/error_messages.hpp"
+
+namespace oneapi::dal::preview::louvain::detail {
+
+template <typename Task>
+class descriptor_impl : public base {
+public:
+    double accuracy_threshold = 0.0001;
+    double resolution = 1;
+    std::int64_t max_iteration_count = 0;
```

Comment:
What if the user does not set this parameter?

Reply:
The algorithm will be executed until the vertices change their community in the local_moves phase

Reply:
what is default for other libs? other ML algos with similar iterative process.

Reply:
SAS - maxiters=20
Louvain - maxLevels=10
cuGraph - max_iter=100
https://github.com/CWTSLeiden/networkanalysis (reference implementation of leiden, louvain is also implemented) - iterations=10

according to the article "From Louvain to Leiden: guaranteeing well-connected communities" - "Louvain quickly converges to a partition and is then unable to make further improvements.". in this article, only the first 10 iterations of the algorithms are considered, since louvain does not demonstrate a significant increase in modularity already on them. 

For datasets with 300k/900k vertices and 1kk/3.7kk edges with accuracy_threshold=0.0001 and resolution=1.0 the implementation of the algorithm was completed for 7 iterations. Therefore, default=10 looks like the optimal value at least for sequential implementation (this value is used in neo4j)